src/core/border_router/routing_manager.cpp

```diff
@@ -1834,6 +1854,91 @@ void RoutingManager::DiscoveredPrefixTable::SignalTableChanged(void)
     mSignalTask.Post();
 }
 
+void RoutingManager::DiscoveredPrefixTable::ProcessNeighborAdvertMessage(
+    const Ip6::Nd::NeighborAdvertMessage &aNaMessage,
+    const Ip6::Address &                  aSrcAddress)
+{
+    Router *router = mRouters.FindMatching(aSrcAddress);
+
+    VerifyOrExit(router != nullptr);
+    VerifyOrExit(aNaMessage.IsRouterFlagSet());
```

Comment:
We are rejecting a NA message which has the `R` unset but we are not sending NA messages on ourselves to respond to NS from another OT Border Router which will likely handle NS by the kernel and unset the `R` flag.

I think we don't require the peer to be a router (the router here is the router for the infra link) to act as a Thread Border Router - Wi-Fi/Ethernet hosts can be a Thread Border Router as well. So what about removing the check here?

Reply:
I am ok relaxing this check. Thoughts? @jwhui ?

> I think we don't require the peer to be a router (the router here is the router for the infra link) to act as a Thread Border Router - Wi-Fi/Ethernet hosts can be a Thread Border Router as well. So what about removing the check here?

I am not sure about this but I think since we receive or send RA from a Thread BR, we expect it to be a router and for it to indicate this in NA as well (has `R` flag set). 

For our BRs, we expect NAs to be sent by kernel. But I recommend we configure it to ensure we include `R` flag in our emitted NA (so to have all other client that may rely on this flag to treat it us a router)

Reply:
Relaxes the check in new push (rebased as well).